persistent-sqlite/Database/Persist/Sqlite.hs

```diff
@@ -265,9 +263,9 @@ wrapConnectionInfo connInfo conn logFunc = do
 -- that all log messages are discarded.
 --
 -- @since 1.1.4
-runSqlite :: (MonadUnliftIO m, IsSqlBackend backend)
+runSqlite :: (MonadUnliftIO m)
           => Text -- ^ connection string
-          -> ReaderT backend (NoLoggingT (ResourceT m)) a -- ^ database action
+          -> ReaderT SqlBackend (NoLoggingT (ResourceT m)) a -- ^ database action
```

Comment:
This is another thing that should be easy to generalize on our side.

Reply:
Well, no, actually :stuck_out_tongue: 

The entire point it that the `backend/SqlBackend` here, is from the `open'` function, which creates the `SqlBackend` in the `withSqliteConn`. Making a `SqlBackend` into a `BackendCompatible SqlBackend backend => backend` would imply creating a `backend` from a `SqlBackend`. That being annoying and undesirable was the entire point.
Remember, this is Sqlite specific, so it's gonna have to work with a `SqlBackend`.

The `RawSqlite` PR should work now afaict, though. Then the `SqlBackend` here would turn into a `RawSqlite SqlBackend` which is what `merijn` was trying to accomplish, I think.

Reply:
That's not quite right - `BackendCompatible SqlBackend backend` implies that you can create a `SqlBackend` from a `backend`. `runSqlite` would be able to run an action that uses `SqlBackend`, or `SqlReadBackend`, or `SqlFor SomeDb`, or `RawSqlite SqlBackend` (provided that was given an `instance BackendCompatible SqlBackend (RawSqlite SqlBackend)`).

The problem with `IsSqlBackend` is that it requires an *isomorphism* - that you can go from a `SqlBackend` to your type. `RawSqlite SqlBackend` has more than a mere `SqlBackend`, so there's no `SqlBackend -> RawSqlite SqlBackend` you can use.

Reply:
We can try the polymorphic `BackendCompatbile SqlBackend backend => backend` stuff and if it presents problems with the raw SQLite stuff then we can revert those functions to concrete types.

It's not a deal breaker either way - requiring `ReaderT SqlBackend ...` just means that library users call `local SqlReadBackend` to map the types how they want instead of the library doing it automatically.

Reply:
I'll try it one more time, maybe I'm just derping with the type system.